src/lib.rs

```diff
@@ -51,8 +51,35 @@ pub trait Color {
 /// or [`BgDynColorDisplay`](BgDynColorDisplay). If your color will be known at compile time it
 /// is recommended you avoid this.
 pub trait DynColor {
-    fn fmt_ansi_fg(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result;
-    fn fmt_ansi_bg(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result;
+
+    // XXX needed for dyn_styles.rs. Should we better return `Result`? It doesn't seem necessary right now, but maybe in the future.
+    fn get_fg(&self) -> DynStylesColor;
+    fn get_bg(&self) -> DynStylesColor;
+
+    // XXX Do you have any idea how to handle this without `DynStylesColor`? I tried half a day, but I couldn't find anything better. Without `dynamic::Rgb` we could use a `&'static str` easily, but when I hit `Rgb` everything because a pain because there wasn't an easy way to format the RGB string. Generating and returning a UTF-8 byte array, instead of `DynStylesColor`, and using `str::from_utf8(&byte_array).unwrap()` was the best I found (at least I think, this should work). But that would make everything else a little less efficient and more complex, so I went with `DynStylesColor`
+    fn fmt_ansi_fg(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if let DynStylesColor::Ansi(ansi) = self.get_fg() {
+            write!(f, "{}", ansi)
+        } else {
+            unreachable!() // RGB based implementations implement this themself
```

Comment:
Could you elaborate on this? I'm a bit confused.

Reply:
The comment means, that this default implementation of `fmt_ansi_fg` is overwritten in "impl DynColor for Rgb", and that the `unreachable!` will never be executed.

Reply:
Correct me if I'm wrong but this is still a public-facing method, and if used directly would cause a panic (not to mention it probably makes the codegen worse)

Reply:
Are you referring to the new code or to the old code above (which doesn't exist anymore)? How could the new version panic?

I didn't realize that it's bad to add public methods to these traits.  I will create new traits that are private to the crate.

Reply:
Ah, that doesn't even work (a private trait in a public API). So what would be the alternative? 

The best I could find was to hide the methods via `#[doc(hidden)]`.